### PR TITLE
NxSegmentedButton RTL conversion- RSC-1363

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://issues.sonatype.org/browse/RSC-1363

### CODE REVIEW

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx [hunk 1]
```diff
@@ -4,207 +4,309 @@
...
+      await act(async () => { await user.click(dropdownToggleBtn); });
+      expect(onClick).not.toHaveBeenCalled();
 
-    act(() => {
-      // Jest/JSDom need actual events to be able to trigger effects properly
-      // See https://stackoverflow.com/questions/27557624/simulating-click-on-document-reactjs-jsdom
-      element.find('button.nx-segmented-btn__dropdown-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
+      await act(async () => { await user.click(mainBtn); });
```
Comment (EmilyBermudez88) [inline lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx:83]: In some situations in the stateful conversion, I was getting the `Warning: An update to ForwardRef(nxStatefulSegmentedButton) inside a test was not wrapped in act(...)`. It usually seemed to happen when closing the dropdown, although it was also necessary in this test.

File: lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx [hunk 2]
```diff
@@ -31,384 +19,321 @@ describe('NxSegmentedButton', function() {
...
-  it('renders nx-segmented-btn__main-btn and nx-segmented-btn__dropdown-btn NxButtons as children', function() {
-    expect(getMounted()).toContainMatchingElement('.nx-segmented-btn__main-btn');
-    expect(getMounted()).toContainMatchingElement('.nx-segmented-btn__dropdown-btn');
-    const mainButton = getMounted().find('ForwardRef(NxButton).nx-segmented-btn__main-btn');
-    expect(mainButton).toMatchSelector(NxButton);
-    expect(getMounted().find('ForwardRef(NxButton).nx-segmented-btn__dropdown-btn')).toMatchSelector(NxButton);
-  });
+    it('passes the type to the main btn if specified', function() {
+      const defaultBtn = quickRender().getByRole('button', { name: 'Click Me' }),
+          customBtn = quickRender({ type: 'button'}).getByRole('button', { name: 'Click Me' }),
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx:63]:
```suggestion
          customBtn = quickRender({ type: 'button' }).getByRole('button', { name: 'Click Me' }),
```

File: lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx [hunk 3]
```diff
@@ -31,384 +19,321 @@ describe('NxSegmentedButton', function() {
...
-
-    expect(onToggleOpen).toHaveBeenCalled();
+    await user.click(document.body);
+    expect(onToggleOpen).toHaveBeenCalledTimes(1);
   });
 
-  it('does not call onToggleOpen if a click happens anywhere when the dropdown is closed', function() {
-    const onToggleOpen = jest.fn(),
-        component = getMounted({ onToggleOpen }, { attachTo: container });
+  it('does not call onToggleOpen if a click happens anywhere aside from the'
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx:177]:
```suggestion
  it('does not call onToggleOpen if a click happens anywhere aside from the '
```

File: lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx [hunk 4]
```diff
@@ -31,384 +19,321 @@ describe('NxSegmentedButton', function() {
...
+        component = renderEl({ onToggleOpen, isOpen: true })!,
+        dropdownBtn = within(component).getByRole('button', { name: 'more options' });
 
-    component.simulate('keyDown', { key: 'Escape', preventDefault: jest.fn() });
+    dropdownBtn.focus();
+    await user.keyboard('{Escape}');
     expect(onToggleOpen).toHaveBeenCalled();
   });
 
   it('calls preventDefault on Escape keydown', function() {
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx:211]: See my recent commits in the `NxFilterDropdown` branch for an elaboration on this.  It is only supposed to call `preventDefault` when the dropdown is open.

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx [hunk 5]
```diff
@@ -4,207 +4,309 @@
...
+
+    await user.click(dropdownToggleBtn);
+    const menu = component.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await act(async () => { await user.click(document.body); });
+    expect(menu).not.toBeInTheDocument();
+  });
+
+  it('closes the dropdown when an outside click happens', async function() {
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx:164]: How is this meaningfully different from the test above?

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx [hunk 6]
```diff
@@ -4,207 +4,309 @@
...
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
+    await user.click(dropdownToggleBtn);
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+  });
+
+  it('does not open when a click happens anywhere aside from the dropdown toggle button'
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx:206]:
```suggestion
  it('does not open when a click happens anywhere aside from the dropdown toggle button '
```

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx [hunk 7]
```diff
@@ -4,207 +4,309 @@
...
+    await user.click(dropdownToggleBtn);
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).toBeInTheDocument();
+
+    await fireEvent.keyDown(menu as HTMLElement, { key: 'Escape' });
+    menu = container.querySelector('.nx-dropdown-menu');
+    expect(menu).not.toBeInTheDocument();
   });
 
+  it('calls preventDefault on Escape keydown', async function() {
```
Comment (rpokorny) [inline lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx:245]: Only when the dropdown is open

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx (hunk 5 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx]: > How is this meaningfully different from the test above?

Yep you're right. Thanks for catching that, removed here: [f6dbd83](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f6dbd8353e17e5fed2d340e366eddefe4fc25421)

File: lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx (hunk 4 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxSegmentedButton/__tests__/NxSegmentedButton.test.tsx]: > See my recent commits in the `NxFilterDropdown` branch for an elaboration on this. It is only supposed to call `preventDefault` when the dropdown is open.

Thanks for pointing that out- updated here: [f25e053](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f25e053572e0e0069e299e9dbef87382cb8484b0)

File: lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx (hunk 7 above)
Comment (EmilyBermudez88) [reply lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx]: > Only when the dropdown is open

[f25e053](https://github.com/sonatype/sonatype-react-shared-components/pull/927/commits/f25e053572e0e0069e299e9dbef87382cb8484b0)

**Review by rpokorny — APPROVED** [review]

**Pull request merged**